### ISSUE TITLE
unit test for deleteGenesisDeposits  should not pass 

### ISSUE THREAD

**Issue by sadiq1971** [ChainSafe/lodestar#233]
**Describe the bug**
In function deleteGenesisDeposits variable criteria is undifined. But it shouldn't be. But the corresponding test is passing.
**Reference**
src
https://github.com/ChainSafe/lodestar/blob/09a28d5112c5a8818d04f041c09bcc5466643af6/src/db/api/beacon/beacon.ts#L219-L224
test
https://github.com/ChainSafe/lodestar/blob/09a28d5112c5a8818d04f041c09bcc5466643af6/test/unit/db/api/beacon.test.ts#L478-L494

**Comment by sadiq1971** [comment]
The corresponding test is only checking whether batchDelete method  is called with argument of type Array. But it should also check the argument value. And also the callback function in deposits.map() should return something or use push operation to criteria. So, the updated code can be ..


src---

``` 
 public async deleteGenesisDeposits(deposits: Deposit[]): Promise<void> {
    const criteria = deposits.map((deposit) => {
       return encodeKey(Bucket.genesisDeposit, deposit.index);
    });
    await this.db.batchDelete(criteria);

  }
```

test
```
it('test delete genesis deposits', async function() {
    encodeKeyStub.returns('genesisDepositKey');
    let argForBatchDelete = ['genesisDepositKey','genesisDepositKey'];
    dbStub.batchDelete.resolves({});
    await beaconDB.deleteGenesisDeposits(
      [generateDeposit(1), generateDeposit(2)]
    );
    expect(
      encodeKeyStub.withArgs(Bucket.genesisDeposit, sinon.match.any).calledTwice
    ).to.be.true;
    expect(
      dbStub.batchDelete.calledWith(argForBatchDelete)
  ).to.be.true;
  });

```

**Comment by mpetrunic** [comment]
Yeah, my bad, method was different when writing tests and sinon is missing matcher for array length.
 You could open PR with those changes :)

**Comment by sadiq1971** [comment]
okay.

**Comment by mpetrunic** [comment]
fixed with #245
